app/assets/javascripts/admin/payments/services/payment.js.coffee

```diff
@@ -42,8 +42,9 @@ angular.module('admin.payments').factory 'Payment', (AdminStripeElements, curren
 
     submit: =>
       munged = @preprocess()
-      PaymentResource.create({order_id: munged.order_id}, munged, (response, headers, status)=>
-        $window.location.pathname = "/admin/orders/" + munged.order_id + "/payments"
+      PaymentResource.create({order_id: munged.order_id}, munged, (response, headers, status) ->
+        document.body.innerHTML = Object.values(response).join('')
+        $window.history.pushState({}, '', "/admin/orders/" + munged.order_id + "/payments")
```

Comment:
I wish I knew enough about angular to suggest how we could update it to use a standard form submission, instead of having to try and simulate it here.

But oh well, I can see how this would work, thanks for fixing this!

Reply:
Might be worth rewriting this area at some point to simplify this. Standard form submission would be better, but this doesn't hurt for now.